# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/GCAPI.h
# Commit: 7da8f779dc24
# Full Hash: 7da8f779dc24489bd423f647c1d60530e9deb24e
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-12-17 21:49:27
# Regressor Bug: 1681533
# File Overlap Count: 2
# Description:
#   Bug 1681533 - Only collect the nursery when necessary in major GC r=sfink
#   
#   This rearranges a few things so that we can run a minor GC inside a major GC
#   slice, so we don't have to be conservative about collecting the nursery at the
#   start of a slice.
# ==============================================================================

diff -r bc20cf2a3ad8 -r 7da8f779dc24 js/public/GCAPI.h
--- a/js/public/GCAPI.h	Thu Dec 17 17:39:25 2020 +0000
+++ b/js/public/GCAPI.h	Thu Dec 17 17:57:33 2020 +0000
@@ -391,6 +391,16 @@
    * string will be allocated in nursery.
    */
   JSGC_STOP_PRETENURE_STRING_THRESHOLD = 43,
+
+  /**
+   * A number that is incremented on every major GC slice.
+   */
+  JSGC_MAJOR_GC_NUMBER = 44,
+
+  /**
+   * A number that is incremented on every minor GC.
+   */
+  JSGC_MINOR_GC_NUMBER = 45,
 } JSGCParamKey;
 
 /*